From 77bd086a5635b7f059bda2c8d787fc19712d3e15 Mon Sep 17 00:00:00 2001
From: Tim Lunn <tim@feathertop.org>
Date: Fri, 29 Sep 2023 10:51:30 +1000
Subject: [PATCH] OPTEE for RV1126

Add RV1126 as target for OP-TEE and add various configs for build.
Also ajusts ramdisk address to clear OPTEE memory region.
---
 arch/arm/mach-rockchip/Kconfig   | 3 ++-
 configs/neu2-io-rv1126_defconfig | 5 +++++
 include/configs/rv1126_common.h  | 2 +-
 3 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-rockchip/Kconfig b/arch/arm/mach-rockchip/Kconfig
index 95f89827e0..b818ab9520 100644
--- a/arch/arm/mach-rockchip/Kconfig
+++ b/arch/arm/mach-rockchip/Kconfig
@@ -379,6 +379,7 @@ config ROCKCHIP_RV1126
 	imply SPL_RAM
 	imply SPL_REGMAP
 	imply SPL_ROCKCHIP_COMMON_BOARD
+	select SPL_OPTEE_IMAGE if SPL_FIT
 	imply SPL_SERIAL
 	imply SPL_SYSCON
 
@@ -453,7 +454,7 @@ config ROCKCHIP_BOOT_MODE_REG
 config ROCKCHIP_OPTEE_BINARY
 	bool "Use a OP-TEE binary provided by Rockchip"
 	depends on SPL_OPTEE_IMAGE
-	default y if ROCKCHIP_RK3036 || ROCKCHIP_RK3128 || ROCKCHIP_RK3288
+	default y if ROCKCHIP_RK3036 || ROCKCHIP_RK3128 || ROCKCHIP_RK3288 || ROCKCHIP_RV1126
 	help
 	  This option enables the usage of vendor-provided OP-TEE binaries.
 	  Say Y if you are using OP-TEE binary provided by Rockchip.
diff --git a/configs/neu2-io-rv1126_defconfig b/configs/neu2-io-rv1126_defconfig
index f02c38c566..2f53600861 100644
--- a/configs/neu2-io-rv1126_defconfig
+++ b/configs/neu2-io-rv1126_defconfig
@@ -14,6 +14,11 @@ CONFIG_DEBUG_UART_CLOCK=24000000
 CONFIG_SYS_LOAD_ADDR=0xe00800
 CONFIG_DEBUG_UART=y
 CONFIG_FIT_VERBOSE=y
+CONFIG_NR_DRAM_BANKS=2
+CONFIG_ROCKCHIP_OPTEE_BINARY=y
+CONFIG_SPL_FIT=y
+CONFIG_SPL_LOAD_FIT=y
+# CONFIG_USE_SPL_FIT_GENERATOR is not set
 CONFIG_DEFAULT_FDT_FILE="rv1126-edgeble-neu2-io.dtb"
 # CONFIG_DISPLAY_CPUINFO is not set
 CONFIG_DISPLAY_BOARDINFO_LATE=y
diff --git a/include/configs/rv1126_common.h b/include/configs/rv1126_common.h
index a64c0c6364..6961dbe20b 100644
--- a/include/configs/rv1126_common.h
+++ b/include/configs/rv1126_common.h
@@ -26,7 +26,7 @@
 	"fdt_addr_r=0x08300000\0" \
 	"fdtoverlay_addr_r=0x02000000\0" \
 	"kernel_addr_r=0x02008000\0" \
-	"ramdisk_addr_r=0x0a200000\0"
+	"ramdisk_addr_r=0x0a400000\0"
 
 #include <config_distro_bootcmd.h>
 #define CFG_EXTRA_ENV_SETTINGS \
-- 
2.40.1

